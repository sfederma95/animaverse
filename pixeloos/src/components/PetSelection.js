import React, {useEffect, useState} from 'react';
import Pet from './Pet';
import rat from '../pets/rat.png'
import panda from '../pets/panda.png'
import cat from '../pets/cat.png'
import bird from '../pets/bird.webp'
import dog from '../pets/dog.png'
import dinosaur from '../pets/dinosaur.png'

function PetSelection({selectPetType}){
    const [petNum, setPetNum] = useState(0)
    const showPet = (n) => {
        const petList = window.jQuery('.adopt-pets');
        window.jQuery(petList[n]).css('display','')
        window.jQuery(petList[n]).siblings('.adopt-pets').css('display','none')
    }
    useEffect(function(){
        showPet(petNum)
    },[petNum])
    const arrowForward = () =>{
        const petList = window.jQuery('.adopt-pets');
        if (petNum > petList.length-1) {
            setPetNum(0)
        }
        else{
            setPetNum(petNum+1)
        }
    }
    const arrowBack= () =>{
        const petList = window.jQuery('.adopt-pets');
        if (petNum === 0) {
            setPetNum(petList.length-1)
        }
        else{
            setPetNum(petNum-1)
        }
    }
    return (
        <div className="slideshow-container">
            <button className="prev" onClick={arrowBack}>&#10094;</button>
            <Pet src={rat} n={1} type='Rat' level={1} buttonFn={selectPetType} />
            <Pet src={panda} n={2} type='Panda' level={1} buttonFn={selectPetType}/>
            <Pet src={cat} n={3} type='Cat' level={1} buttonFn={selectPetType}/>
            <Pet src={dinosaur} n={4} type='Dinosaur' level={1} buttonFn={selectPetType}/>
            <Pet src={bird} n={5} type='Bird' level={1} buttonFn={selectPetType}/>
            <Pet src={dog} n={6} type='Dog' level={1} buttonFn={selectPetType}/>
            <button className="next" onClick={arrowForward}>&#10095;</button>
        </div>
    )
}

export default PetSelection;