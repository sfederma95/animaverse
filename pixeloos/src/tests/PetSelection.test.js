import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import PetSelection from '../components/PetSelection';
import {UserProvider} from './testSetup';

it ('renders pet select component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <PetSelection />
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <PetSelection />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})