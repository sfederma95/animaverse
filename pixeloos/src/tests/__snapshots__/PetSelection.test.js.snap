// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="slideshow-container"
  >
    <button
      class="prev"
    >
      ❮
    </button>
    <div
      class="adopt-pets"
    >
      <p>
        1 / 6
      </p>
      <img
        alt="Rat"
        src="rat.png"
      />
      <p>
        Type: Rat
      </p>
      <p>
        Level: 1
      </p>
    </div>
    <div
      class="adopt-pets"
      style="display: none;"
    >
      <p>
        2 / 6
      </p>
      <img
        alt="Panda"
        src="panda.png"
      />
      <p>
        Type: Panda
      </p>
      <p>
        Level: 1
      </p>
    </div>
    <div
      class="adopt-pets"
      style="display: none;"
    >
      <p>
        3 / 6
      </p>
      <img
        alt="Cat"
        src="cat.png"
      />
      <p>
        Type: Cat
      </p>
      <p>
        Level: 1
      </p>
    </div>
    <div
      class="adopt-pets"
      style="display: none;"
    >
      <p>
        4 / 6
      </p>
      <img
        alt="Dinosaur"
        src="dinosaur.png"
      />
      <p>
        Type: Dinosaur
      </p>
      <p>
        Level: 1
      </p>
    </div>
    <div
      class="adopt-pets"
      style="display: none;"
    >
      <p>
        5 / 6
      </p>
      <img
        alt="Bird"
        src="bird.webp"
      />
      <p>
        Type: Bird
      </p>
      <p>
        Level: 1
      </p>
    </div>
    <div
      class="adopt-pets"
      style="display: none;"
    >
      <p>
        6 / 6
      </p>
      <img
        alt="Dog"
        src="dog.png"
      />
      <p>
        Type: Dog
      </p>
      <p>
        Level: 1
      </p>
    </div>
    <button
      class="next"
    >
      ❯
    </button>
  </div>
</DocumentFragment>
`;
